fission-web-server/library/Fission/Web/Server/Authorization/Types.hs

```diff
@@ -27,7 +27,7 @@ data Heroku = Heroku
 data Authorization = Authorization
   { sender   :: Either Heroku DID
   , about    :: Entity User
-  , potency  :: Potency
+  , potency  :: Maybe Potency -- Nothing indicates "just used for signature checking"
```

Comment:
Rather than dealing with the `Maybe` for all possible call sites in the future, could we consider either abstracting over `Potency`, or parsing that case to a different data type? It's not usable if only checking the signature, no? This type is specialized to look at the database with `User`, so it's not a general use thing.

Reply:
> Let's make error cases as unrepresentable as possible, if possible!
The new `Potency` type lacks a variant the old one had before: `AuthNOnly`. So basically `Old.Potency` is isomorphic to `Maybe New.Potency`. So I don't think there should be any changes in terms of what's representable. I basically moved the comment for `Nothing` over from the definition of the old `Potency` type.

---

In any way, as soon as hs-ucan is updated to UCAN 0.8 and I'm incorporating that into the fission stuff, this will change.